src/main/java/de/zalando/aruha/nakadi/config/RepositoriesConfig.java

```diff
@@ -36,15 +38,25 @@
     private ZookeeperConfig zookeeperConfig;
 
     @Bean
-    public EventTypeRepository eventTypeRepository() throws Exception {
-        final EventTypeRepository dbRepo = new EventTypeDbRepository(jdbcTemplate, jsonConfig.jacksonObjectMapper());
-
+    public EventTypeCache eventTypeCache() throws Exception {
         final String connectString = zookeeperConfig.zooKeeperHolder().get().getZookeeperClient().getCurrentConnectionString();
         final RetryPolicy retryPolicy = new ExponentialBackoffRetry(1000, 3);
         final CuratorFramework client = CuratorFrameworkFactory.newClient(connectString, retryPolicy);
 
         client.start();
 
-        return new EventTypeCachedRepository(dbRepo, new EventTypeCache(dbRepo, client));
+        EventBodyMustRespectSchema strategy = new EventBodyMustRespectSchema();
+        ValidationStrategy.register(EventBodyMustRespectSchema.NAME, strategy);
+
+        return new EventTypeCache(dbRepo(), client);
+    }
+
+    @Bean
+    public EventTypeRepository eventTypeRepository() throws Exception {
+        return new EventTypeCachedRepository(dbRepo(), eventTypeCache());
+    }
+
+    private EventTypeRepository dbRepo() {
```

Comment:
I would also suggest giving it a meaningful name like `eventTypeRepository`.
However, you then have two `EventTypeRepository` bean which will lead to trouble when using `@Autowired`. Hence, it would be good having The `DatabaseEventTypeRepository` not as a spring bean. That leads me to the question why `EventTypeCachedRepository` needs a `EventTypeCache` and both beans need a 'EventTypeRepository'.